### PR TITLE
Logging

### PR DESCRIPTION
**Pull request by afugazzotto**
Fixes https://github.com/numaproj-labs/numaplane/issues/127

### Modifications

Replaced the 2 logging libraries (zap and logrus) with a single library (zerolog). This decision was made to consolidate multiple logging utilities into one, standardize log output, allow for future features, and possible to allow reusability of this logger with other numaproj tools. Tried to keep older functionality unchanged as much as possible. 

### Verification

Tested the changes manually by inspecting the logs and also wrote unit tests with 87.7% coverage.

### CODE REVIEW

**Comment by juliev0** [comment]
if @xdevxy is good with this, I don't need to review as well..thanks!

File: cmd/agent/main.go [hunk 1]
```diff
@@ -20,12 +20,12 @@ import (
...
 
 	"github.com/numaproj-labs/numaplane/internal/agent"
-	"github.com/numaproj-labs/numaplane/internal/util/logging"
+	"github.com/numaproj-labs/numaplane/internal/util/logger"
 )
 
 var (
 	// logger is the global logger for the numaplane-agent
-	logger     = logging.NewLogger().Named("numaplane-agent")
+	numaLogger = logger.New(nil, nil).WithName("numaplane-agent")
```
Comment (xdevxy) [inline cmd/agent/main.go:28]: As discussed add a new method `NewWithWriter` and keep this one (`New`) without any parameters if we don't need them.

File: internal/util/logger/logger.go [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+	zerolog.SetGlobalLevel(logrVerbosityToZerologLevelMap[verboseLevel])
+
+	w := io.Writer(os.Stdout)
+	if writer != nil {
+		w = *writer
+	}
+
+	lvl := int(defaultLevel)
+	if level != nil {
+		lvl = *level
```
Comment (xdevxy) [inline internal/util/logger/logger.go:88]: I don't follow how this `level` is used? and it is default to debug level if not provided?

Comment (afugazzotto) [reply internal/util/logger/logger.go:96]: `level` is the argument to the function. If it is omitted, the function will use a default level (I decided to use Info level for default).

File: internal/common/common.go [hunk 3]
```diff
@@ -5,8 +5,6 @@ const (
 	AnnotationKeyGitSyncInstance = "numaplane.numaproj.io/tracking-id"
 	// SSAManager is the default numaplane manager name used by server-side apply syncs
 	SSAManager = "numaplane-controller"
-	// EnvLogFormat log format that is defined by `--logformat` option
-	EnvLogFormat = "NUMAPLANE_LOG_FORMAT"
 	// EnvLogLevel log level that is defined by `--loglevel` option
```
Comment (xdevxy) [inline internal/common/common.go:8]: nit: is this comment still hold true? e.g. we still have`--loglevel` ?

**Review by xdevxy — APPROVED** [review]

File: internal/common/common.go (hunk 3 above)
Comment (afugazzotto) [reply internal/common/common.go:8]: I removed the NUMAPLANE_LOG_FORMAT because it no longer applies.

**Pull request merged**